File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Awareness.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.LuaInterface;

@LuaInterface


public class Awareness extends FlavourBuff {

	public static final float DURATION = 2f;

	@Override
	public void detach() {
		super.detach();
		target.observe();
	}
}

File: scripts/buffs/Awareness.lua
---
--- Awareness: item searching; view recalculated on detach
--- (was actors/buffs/Awareness.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

return buff.init{
    desc  = function ()
        return {
            icon          = -1, -- no indicator icon
            name          = "AwarenessBuff_Name",
            info          = "AwarenessBuff_Info",
        }
    end,

    detach = function(self, buff)
        buff.target:observe()
    end
}
